middleware/auth.js

```diff
@@ -0,0 +1,152 @@
+const passport = require('passport')
+const localStrategy = require('passport-local').Strategy
+const CustomStrategy = require('passport-custom').Strategy
+const jwt = require('jsonwebtoken')
+const UserModel = require('../models/user.js')
+const nodemailer = require('nodemailer')
+
+passport.use(
+  'signup',
+  new localStrategy(
+    {
+      usernameField: 'email',
+      passwordField: 'password',
+      passReqToCallback: true,
+    },
+    async (req, email, password, done) => {
+
+      const { firstName, lastName, profile_image } = req.body
+      try {
+        let user;
+        user = await UserModel.findOne({ email })
+        if (user) {
+          return done(null, false, { message: 'User already exists with this email' })
+        }
+        user = await UserModel.create({ firstName, lastName, profile_image, email, password, profile_image })
```

Comment:
Encrypt the password using bcrypt before storing in db.

Reply:
encrypted string is stored into the db already. The helper function is in the models file.